### PR TITLE
Add default information to CronJobSpec

### PR DESCRIPTION
**Pull request by soltysh**
This is requested followup to https://github.com/kubernetes/kubernetes/pull/41890

@smarterclayton @janetkuo ptal

### CODE REVIEW

**Review by janetkuo — APPROVED** [review]

**Comment by janetkuo** [comment]
@k8s-bot bazel test this
@k8s-bot cvm gce e2e test this

**Comment by smarterclayton** [comment]
Does the cron job fuzzer need to be changed to handle the defaulting?

**Comment by soltysh** [comment]
> Does the cron job fuzzer need to be changed to handle the defaulting?

It [already does that](https://github.com/kubernetes/kubernetes/blob/c3d7ee11d176d94108831bed2e76e94039338fd2/pkg/api/testing/fuzzer.go#L215-L217) this is only surfacing this information to users through doc update.

**Comment by janetkuo** [comment]
Updated title so that it's clear we're just updating information, not updating default values.

**Comment by soltysh** [comment]
@smarterclayton mind approving this one, since this is only docs PR.

**Comment by enisoc** [comment]
I'm removing the v1.6 milestone since we are lifting code freeze soon and it doesn't seem like we need to hold the branch cut for this. Please follow the [cherrypick procedure](https://github.com/kubernetes/community/blob/master/contributors/devel/cherry-picks.md) if you still think this should go into the release-1.6 branch.

**Comment by soltysh** [comment]
Sure, will do so. Thx.

File: pkg/apis/batch/v2alpha1/types_swagger_doc_generated.go [hunk 1]
```diff
@@ -52,7 +52,7 @@ var map_CronJobSpec = map[string]string{
 	"":                           "CronJobSpec describes how the job execution will look like and when it will actually run.",
 	"schedule":                   "Schedule contains the schedule in Cron format, see https://en.wikipedia.org/wiki/Cron.",
 	"startingDeadlineSeconds":    "Optional deadline in seconds for starting the job if it misses scheduled time for any reason.  Missed jobs executions will be counted as failed ones.",
-	"concurrencyPolicy":          "ConcurrencyPolicy specifies how to treat concurrent executions of a Job.",
+	"concurrencyPolicy":          "ConcurrencyPolicy specifies how to treat concurrent executions of a Job. Defaults to Allow.",
```
Comment (bgrant0607) [inline pkg/apis/batch/v2alpha1/types_swagger_doc_generated.go]: @soltysh PTAL. This is user-facing documentation. Only json field names, not Go field names, should appear in the documentation. Please fix this in types.go

Comment (soltysh) [inline pkg/apis/batch/v2alpha1/types_swagger_doc_generated.go]: I've noticed we don't have this addressed anywhere in batch/types.go. I'll open another PR addressing this in the entire batch package as a followup if you don't mind.

**Comment by soltysh** [comment]
@k8s-bot non-cri e2e test this

File: pkg/apis/batch/v2alpha1/types_swagger_doc_generated.go (hunk 1 above)
Comment (soltysh) [inline pkg/apis/batch/v2alpha1/types_swagger_doc_generated.go]: I've created https://github.com/kubernetes/kubernetes/pull/44038 to address that.

**Comment by soltysh** [comment]
@k8s-bot kubemark e2e test this

@k8s-bot gce etcd3 e2e test this

**Comment by soltysh** [comment]
@k8s-bot node e2e test this

**Comment by janetkuo** [comment]
@soltysh do we need to cherrypick this into 1.6?

**Pull request merged**